src/test/java/fr/esgi/al5_2/Tayarim/ProprietaireServiceTest.java

```diff
@@ -26,262 +25,281 @@
+    LocalDateTime localDateTime = LocalDateTime.now();
+
+    Proprietaire proprietaire = Proprietaire.builder()
+        .nom("Ferreira")
+        .prenom("Mathieu")
+        .email("test@gmail.com")
+        .numTel("0612345678")
+        .motDePasse("password")
+        .dateInscription(localDateTime)
+        .isPasswordUpdated(Boolean.TRUE)
+        .build();
+    proprietaire.setLogements(null);
+    proprietaire.setId(1L);
+    when(proprietaireRepository.findById(proprietaire.getId())).thenReturn(
+        Optional.of(proprietaire));
+
+    ProprietaireDto proprietaireDTO = proprietaireService.getProprietaireById(proprietaire.getId(),
+        false);
+
+    Assertions.assertThat(proprietaireDTO).isNotNull();
+    Assertions.assertThat(proprietaireDTO.getId()).isEqualTo(proprietaire.getId());
+    Assertions.assertThat(proprietaireDTO.getNom()).isEqualTo(proprietaire.getNom());
+    Assertions.assertThat(proprietaireDTO.getPrenom()).isEqualTo(proprietaire.getPrenom());
+    Assertions.assertThat(proprietaireDTO.getEmail()).isEqualTo(proprietaire.getEmail());
+    Assertions.assertThat(proprietaireDTO.getNumTel()).isEqualTo(proprietaire.getNumTel());
+    Assertions.assertThat(proprietaireDTO.getDateInscription())
+        .isEqualTo(proprietaire.getDateInscription());
+
+  }
+
+  @Test
+  public void ProprietaireService_GetProprietaireByEmail_ReturnOneProprietaireDto() {
+    LocalDateTime localDateTime = LocalDateTime.now();
+
+    Proprietaire proprietaire = Proprietaire.builder()
+        .nom("Ferreira")
+        .prenom("Mathieu")
+        .email("test@gmail.com")
+        .numTel("0612345678")
+        .motDePasse("password")
+        .dateInscription(localDateTime)
+        .isPasswordUpdated(Boolean.TRUE)
+        .build();
+    proprietaire.setLogements(null);
+    proprietaire.setId(1L);
+    when(proprietaireRepository.findFirstByEmail(proprietaire.getEmail())).thenReturn(
+        Optional.of(proprietaire));
+
+    ProprietaireDto proprietaireDTO = proprietaireService.getProprietaireByEmail(
+        proprietaire.getEmail());
+
+    Assertions.assertThat(proprietaireDTO).isNotNull();
+    Assertions.assertThat(proprietaireDTO.getId()).isEqualTo(proprietaire.getId());
+    Assertions.assertThat(proprietaireDTO.getNom()).isEqualTo(proprietaire.getNom());
+    Assertions.assertThat(proprietaireDTO.getPrenom()).isEqualTo(proprietaire.getPrenom());
+    Assertions.assertThat(proprietaireDTO.getEmail()).isEqualTo(proprietaire.getEmail());
+    Assertions.assertThat(proprietaireDTO.getNumTel()).isEqualTo(proprietaire.getNumTel());
+    Assertions.assertThat(proprietaireDTO.getDateInscription())
+        .isEqualTo(proprietaire.getDateInscription());
+
+  }
+
+  @Test
+  public void ProprietaireService_UpdateProprietaire_ReturnOneProprietaireDto() {
+    LocalDateTime localDateTime = LocalDateTime.now();
+    Long id = 1L;
+    ProprietaireCreationDto proprietaireCreationDTO = ProprietaireCreationDto.builder()
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'proprietaireCreationDTO' declaration and its first usage is 10, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).